### ISSUE TITLE
Reporting for multi-line comments

### ISSUE THREAD

**Issue by GH-Syn** [microsoft/pyright#5192]
## Description

Pyright reports:

`unsupported escape sequence in string literal [reportInvalidStringEscapeSequence]`

when a backslash is followed by an invalid character literal (such as an asterisk) in [multi-line comments](#expected-behavior).  

In multi-line documentation strings, it would appear that there is also [no way to suppress this error](#examples).

## To Reproduce

1. Define any multi-line comment / docstring

```python
def foo():
    """
    The five leaping wizards ... \* They jumped quickly.
    ...
    """
``` 
> It can be in a local (function) scope or in a global scope. In my case, it was found in a local scope

2. Observe the error.

## Expected behavior

Pyright should ignore documentation strings **with the _exception_** of python code contained in doc-strings. Pyright *should* be linting this code:
```python
def foo():
    """
    >>> print("Pyright should lint this code...")
    """
```
> This is the only time pyright should be linting code in documentation strings.

## Examples

See code that produces the error [here](#to-reproduce)

These are some failed workaround attempts at getting the error to go away:

```python
... \* They jumped quickly. # pyright: ignore
```
> This will not suppress anything

```python
    """The five leaping wizards ... \* They jumped quickly.
    ...
    """ # pyright: ignore
```
> Nor will this...

## VS Code extension or command-line

I am running pyright through neovim. It is being (technically) run as a command invoked by my text editor.
```bash
/home/<user>/.local/share/nvim/mason/bin/pyright-langserver --stdio
```
> OS: arch linux btw (sorry) :sweat_smile:
> 
> pyright version: 1.1.310

## Additional comments

Assuming this is a feature, and not a bug:

If this is a feature that I somehow missed, then there should *at least* be the option to add an ignore tag, to mitigate this edge case for those that need to document their code using a library such as Sphinx.

Pyright shouldn't be raising errors in documentation strings, as documentation strings.
Especially when used for sphinx, _as in my case_, because sometimes documentation strings need wacky formatting that python doesn't understand. Such as `rst` which contains backslashes to ignore the next character.

**Comment by erictraut** [comment]
This is a feature. Pyright is reporting that you're using an invalid escape sequence within your string. Refer to the [Python documentation for details](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals). Here's the relevant quote:

> Unrecognized escape sequences produce a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning). In a future Python version they will be a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) and eventually a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError).

If you want your code to continue to run in the future, you should fix these issues. You can include backslashes in your comments, but they must be properly escaped (i.e. as a double backslash).

If you want to ignore the issue for now, you can disable the `reportInvalidStringEscapeSequence` diagnostic check in pyright — either globally for your project or on a per-file basis using a `# pyright: reportInvalidStringEscapeSequence=false` comment at the top of the file.

**Comment by GH-Syn** [comment]
> This is a feature. Pyright is reporting that you're using an invalid escape sequence within your string. Refer to the [Python documentation for details](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals). Here's the relevant quote:
> 
> > Unrecognized escape sequences produce a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning). In a future Python version they will be a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) and eventually a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError).
> 
> If you want your code to continue to run in the future, you should fix these issues. You can include backslashes in your comments, but they must be properly escaped (i.e. as a double backslash).
> 
> If you want to ignore the issue for now, you can disable the `reportInvalidStringEscapeSequence` diagnostic check in pyright — either globally for your project or on a per-file basis using a `# pyright: reportInvalidStringEscapeSequence=false` comment at the top of the file.
> 
> Another solution is to use a "raw" string by preceding the docstring's triple quotes with an `r` character.

My bad. I can see I didn't do enough research looking into this before writing this. Sorry for wasting your time. I'll try to respect the time of maintainers a bit more heading into the future.
